billy-portugal/src/main/java/com/premiumminds/billy/portugal/services/documents/PTGenericInvoiceIssuingHandler.java

```diff
@@ -127,6 +126,9 @@ protected <D extends AbstractDAOPTGenericInvoice<T>> T issue(final T document, f
                                                                 + " does not have a series unique code specified"))
             .toString();
 
+        ZoneId timezone = document.getBusiness().getTimezone();
+        LocalDate issueLocalDate = document.getLocalDate().orElse(LocalDate.ofInstant(invoiceDate.toInstant(), timezone));
```

Comment:
Throw exception in these handlers when localDate is null (e.g. DocumentIssuingException)
Do the same for all country handlers